pkg/images/images_test.go

```diff
@@ -0,0 +1,337 @@
+)
+
+func Test_toKeep(t *testing.T) {
+	type args struct {
+		info types.K8sVersionInfo
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "invalid semVer",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "2.6c.0",
+				MaxRancherVersion:       "",
+				DeprecateRancherVersion: "",
+			}},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "deprecated < 2.7.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "",
+				MaxRancherVersion:       "",
+				DeprecateRancherVersion: "2.6.99",
+			}},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "deprecated = 2.7.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "",
+				MaxRancherVersion:       "",
+				DeprecateRancherVersion: "2.7.0",
+			}},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "2.7.0 < deprecated < 2.8.0 ",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "",
+				MaxRancherVersion:       "",
+				DeprecateRancherVersion: "2.7.2",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "2.7.0 < deprecated < 2.8.0 - 2",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "",
+				MaxRancherVersion:       "",
+				DeprecateRancherVersion: "2.7.5-alpha1",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "deprecated = 2.8.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "",
+				MaxRancherVersion:       "",
+				DeprecateRancherVersion: "2.8.0",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "min < 2.7.0 < 2.8.0 < max",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "2.6.12-alpha2",
+				MaxRancherVersion:       "2.8.2",
+				DeprecateRancherVersion: "",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "min < 2.7.0 < max = 2.8.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "2.6.12-alpha2",
+				MaxRancherVersion:       "2.8.0",
+				DeprecateRancherVersion: "",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "min < 2.7.0 < max < 2.8.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "2.6.12-alpha2",
+				MaxRancherVersion:       "2.7.12-patch1",
+				DeprecateRancherVersion: "",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "min = 2.7.0 < max < 2.8.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "2.7.0",
+				MaxRancherVersion:       "2.7.5",
+				DeprecateRancherVersion: "",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "min = 2.7.0 < max = 2.8.0",
+			args: args{info: types.K8sVersionInfo{
+				MinRancherVersion:       "2.7.0",
+				MaxRancherVersion:       "2.8.0",
+				DeprecateRancherVersion: "",
+			}},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "2.7.0 < min < max < 2.8.0",
```

Comment:
Might need test case like `min <= 2.8.0 < 2.8.99 <= max` to test the upper limit.

Reply:
test case is added.